services/zone-service.ts

```diff
@@ -117,6 +122,82 @@ class ZoneService {
         return await zoneSchema.deleteOne({_id: zoneID});
     }
 
+    async editZone(zoneID: string | string[], newZone: InterestZone): Promise<InterestZone> {
+        let interestZone = await this.findById(zoneID);
+
+        await this.validateZoneEditting(newZone);
+
+        interestZone.address = {
+            ...interestZone.address,
+            name: newZone.address.name
+        };
+        interestZone.status = newZone.status;
+        interestZone.volunteerName = newZone.volunteerName;
+        interestZone.observations = newZone.observations;
+        interestZone.noUnsterilizedCats = newZone.noUnsterilizedCats;
+        interestZone.contactPerson = newZone.contactPerson;
+        
+        await interestZone.save();
+        return interestZone;
+    }
+
+    async deleteCat(zoneID: string | string[], catID: string | string[]): Promise<number> {
+        const interestZone = await this.findById(zoneID);
+        let cat = this.findCat(interestZone, catID, true);
+        if (!cat) {
+            cat = this.findCat(interestZone, catID, false);
+            if (cat) interestZone.noUnsterilizedCats--;
```

Comment:
I am not sure we want to decrease the `noUnsterilizedCats` value, if an invalid cat id was provided